pro/src/api/hooks/useAdapter.ts

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from 'react'
+
+const useAdapter = <TSuccess, TFailure>(
+  getData: () => Promise<AdapterSuccess<TSuccess> | AdapterFailure<TFailure>>,
+  defaultValue: any
+): {
+  data: TSuccess
+  isLoading: boolean
+  error?: string
+} => {
+  const [data, setData] = useState<TSuccess>(defaultValue)
+  const [error, setError] = useState<string>('')
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+  useEffect(() => {
+    async function loadData() {
+      const response = await getData()
+      if (response.isOk) {
+        setData(response.payload)
+      } else {
+        setError(response.message)
+      }
+
+      setIsLoading(false)
+    }
+    if (isLoading) {
+      loadData()
+    }
+  }, [isLoading])
+
+  return { data, isLoading, error }
```

Comment:
```suggestion
  return [data, isLoading, error]
```
Comme suggéré par Marie, c'est plus simple que de devoir connaitre les noms des clés du hash

Reply:
Pourquoi pas, je préfère les clef nommé mais tant pis.